apps/slackbot/task_agent.py

```diff
@@ -150,6 +152,29 @@ def run(self, task: str) -> str:
             action = self.output_parser.parse(assistant_reply)
             print("action:", action)
             tools = {t.name: t for t in self.tools}
+            if action == previous_action:
+                if action.name == "Search" or action.name == "Context Search":
```

Comment:
I recommend we drop this condition since we may want to check duplication for all tools. (We don't need to worry about the `finish` tool for now as it will be separated and it will also break the loop)